### ISSUE TITLE
Not exporting settings in the Top/Bottom category

### ISSUE THREAD

**Issue by Ghostkeeper** [5axes/ImportExportProfiles#2]
When exporting the current settings, the settings from the top/bottom category are missing, both in the Arachne release and in Cura 4.9, using release V1.0.5 of this plug-in.

Steps to reproduce:

1. Add a new Ultimaker 2+ printer. So it'll be the defaults.
2. Change the Top Layers setting to 9.
3. Save this as a custom profile.
4. Go to Extensions -> Import Export Profiles -> Export Current Profile.
5. Choose any file name and save the file.
6. Open the file with a text editor, or a spreadsheet editor like Calc.

The file then contains the following contents:
```
Section;Extruder;Key;Type;Value
general;0;Date;str;08/04/2021 10:56:46
general;0;Os;str;Linux #54-Ubuntu SMP Fri Mar 19 14:25:20 UTC 2021
general;0;Cura_Version;str;4.9.0-BETA
general;0;Profile;str;Customised
general;0;Quality;str;Fine
general;0;Extruder_Count;int;1
resolution;1;layer_height;float;0.1
resolution;1;layer_height_0;float;0.27
resolution;1;line_width;float;0.35
resolution;1;wall_line_width;float;0.35
resolution;1;wall_line_width_0;float;0.35
resolution;1;wall_line_width_x;float;0.35
resolution;1;skin_line_width;float;0.35
resolution;1;infill_line_width;float;0.35
resolution;1;skirt_brim_line_width;float;0.35
resolution;1;initial_layer_line_width_factor;float;100
shell;1;wall_thickness;float;1.05
shell;1;wall_line_count;int;3
shell;1;wall_0_wipe_dist;float;0.2
shell;1;wall_0_inset;float;0.025
shell;1;optimize_wall_printing_order;bool;False
shell;1;outer_inset_first;bool;False
shell;1;alternate_extra_perimeter;bool;False
shell;1;travel_compensate_overlapping_walls_enabled;bool;True
shell;1;travel_compensate_overlapping_walls_0_enabled;bool;True
shell;1;travel_compensate_overlapping_walls_x_enabled;bool;True
shell;1;wall_min_flow;float;0
shell;1;fill_perimeter_gaps;enum;everywhere
shell;1;filter_out_tiny_gaps;bool;True
shell;1;fill_outline_gaps;bool;False
shell;1;xy_offset;float;0
shell;1;xy_offset_layer_0;float;0
shell;1;hole_xy_offset;float;0
shell;1;z_seam_type;enum;sharpest_corner
shell;1;z_seam_corner;enum;z_seam_corner_inner
```

However, doing the same in Cura 4.8 results in the following file:
```
Section;Extruder;Key;Type;Value
general;0;Date;str;08/04/2021 10:55:43
general;0;Os;str;Linux #54-Ubuntu SMP Fri Mar 19 14:25:20 UTC 2021
general;0;Cura_Version;str;4.8.0
general;0;Profile;str;Customised
general;0;Quality;str;Fine
general;0;Extruder_Count;int;1
resolution;1;layer_height;float;0.1
resolution;1;layer_height_0;float;0.27
resolution;1;line_width;float;0.35
resolution;1;wall_line_width;float;0.35
resolution;1;wall_line_width_0;float;0.35
resolution;1;wall_line_width_x;float;0.35
resolution;1;skin_line_width;float;0.35
resolution;1;infill_line_width;float;0.35
resolution;1;skirt_brim_line_width;float;0.35
resolution;1;initial_layer_line_width_factor;float;100
shell;1;wall_thickness;float;1.05
shell;1;wall_line_count;int;3
shell;1;wall_0_wipe_dist;float;0.2
shell;1;roofing_layer_count;int;0
shell;1;top_bottom_thickness;float;0.8
shell;1;top_thickness;float;0.8
shell;1;top_layers;int;9
shell;1;bottom_thickness;float;0.8
shell;1;bottom_layers;int;8
shell;1;initial_bottom_layers;int;8
shell;1;top_bottom_pattern;enum;lines
shell;1;top_bottom_pattern_0;enum;lines
shell;1;skin_angles;[int];[ ]
shell;1;wall_0_inset;float;0.025
shell;1;optimize_wall_printing_order;bool;False
shell;1;outer_inset_first;bool;False
shell;1;alternate_extra_perimeter;bool;False
shell;1;travel_compensate_overlapping_walls_enabled;bool;True
shell;1;travel_compensate_overlapping_walls_0_enabled;bool;True
shell;1;travel_compensate_overlapping_walls_x_enabled;bool;True
shell;1;wall_min_flow;float;0
shell;1;fill_perimeter_gaps;enum;everywhere
shell;1;filter_out_tiny_gaps;bool;True
shell;1;fill_outline_gaps;bool;False
shell;1;xy_offset;float;0
shell;1;xy_offset_layer_0;float;0
shell;1;hole_xy_offset;float;0
shell;1;z_seam_type;enum;sharpest_corner
shell;1;z_seam_corner;enum;z_seam_corner_inner
shell;1;skin_no_small_gaps_heuristic;bool;False
shell;1;skin_outline_count;int;1
shell;1;ironing_enabled;bool;False
shell;1;skin_overlap;float;5
shell;1;skin_overlap_mm;float;0.0175
```

Based on the release notes of V1.0.5, I would expect that the `top_layers` setting is present in the export from the Arachne edition. But all of the top/bottom settings are missing from the export in that release.

Looking at the log file of Cura reveals that this might just be a programming error. This is a snippet of the log from around the time when I exported the file:

```
2021-04-08 10:56:46,117 - ERROR - [MainThread] UM.Logger.logException [112]: Exception: Could not export profile to the selected file
2021-04-08 10:56:46,119 - ERROR - [MainThread] UM.Logger.logException [116]: Traceback (most recent call last):
2021-04-08 10:56:46,120 - ERROR - [MainThread] UM.Logger.logException [116]:   File "/home/trin/.local/share/cura/4.9/plugins/ImportExportProfiles/ImportExportProfiles/ImportExportProfiles.py", line 142, in exportData
2021-04-08 10:56:46,121 - ERROR - [MainThread] UM.Logger.logException [116]:     if VersC > 4.8:
2021-04-08 10:56:46,123 - ERROR - [MainThread] UM.Logger.logException [116]: NameError: name 'VersC' is not defined
```

Looking at that, it's indeed quite unexpected that the rest of the categories also don't show up. Maybe you meant it to say `self.VersC` instead of referencing a local variable.

**Comment by 5axes** [comment]
ok solved 1.0.6

**Comment by Ghostkeeper** [comment]
Tested the new version 1.0.7 and it works. Thanks for the fix!
